pgraph/actions.go

```diff
@@ -414,10 +415,26 @@ func (g *Graph) Worker(v *Vertex) error {
 						playback = true
 						log.Printf("%s[%s]: CheckApply errored: %v", v.Kind(), v.GetName(), e)
 						if retry == 0 {
+							if obj.Prometheus() != nil {
+								if err := obj.Prometheus().UpdateState(fmt.Sprintf("%v[%v]", v.Kind(), v.GetName()), v.Kind(), prometheus.ResStateHardFail); err != nil {
+									// TODO: error correctly
+									log.Printf("%s[%s]: Prometheus.UpdateState() errored: %v", v.Kind(), v.GetName(), err)
+									return
```

Comment:
prom. should be running around the existing code and 
                    instrumenting it, but i would assume it shouldn't be 
                    interferring with behavior or things like that... right?